R/BenchmarkResult.R

```diff
@@ -552,7 +552,9 @@ BenchmarkResult = R6Class("BenchmarkResult",
 as.data.table.BenchmarkResult = function(x, ..., hashes = FALSE, predict_sets = "test", task_characteristics = FALSE) { # nolint
   assert_flag(task_characteristics)
   tab = get_private(x)$.data$as_data_table(view = NULL, predict_sets = predict_sets)
-  tab = tab[, c("uhash", "task", "learner", "resampling", "iteration", "prediction"), with = FALSE]
+  cns = c("uhash", "task", "learner", "resampling", "iteration", "prediction")
+  if ("data_extra" %in% names(tab)) cns = c(cns, "data_extra")
```

Comment:
Maybe instead something like
```suggestion
  cns = c("uhash", "task", "learner", "resampling", "iteration", "prediction",
    if ("data_extra" %in% names(tab)) "data_extra")
```
(same story further down)